### ISSUE TITLE
Azure Advisor Opmization Workbook New Ideas request

### ISSUE THREAD

**Issue by rithinkumark1** [microsoft/finops-toolkit#733]
## 📝 Scenario
I am part of the Finops Team here at KPMG, and we extensively use the optimization workbook for our operations. We have identified some new ideas which would provide more value to the optimization workbook and will help us fulfill our daily task much effectively.
As a **<TODO: What persona is this for?>**, I need to **<TODO: What action does the user need to perform?>** in order to **<TODO: What is the goal? Why is this needed?>**

## 💎 Solution
Ideas.

1. Flexibility to setup alerts in Optimization workbook. The goal is that we can setup email alert so that when a new orphaned resources/ recommendation is added to the workbook we can get an alert in our inbox. Right we need to manually check  everyday if any new resources are added.

2.  Cost Details.
Cost breakout by orphaned resource category: It would be a nice feature to include the cost information for the orphaned resources. Right now we need to manually check the details in the cost management tool.
Cost Anomaly 
It would be helpful if Cost Anomaly alerts were aggregated into a central view, such as a table. right need to manually check insights in each subscriptions.

3. Addition of New Services:  It would be helpful to expand the services covered in the workbook. 
Some of the services which are looking at is 
- Azure Databricks 
- Cognitive Services
- Azure Data Factory v2
- Defender for cloud (existing workbook only provide cost for different plans and total cost)
- Application insights etc.

4. Rightsizing Recommendations across services:
- SQL Database rightsizing based on current utilization. Right now we can see we have only Idle SQL databases
- Insights about SQL Database retention policies, similar to Recovery Services Vault in Backup Center
- Managed disk capacity under-utilization to reduce cost
- Rightsizing recommendations for minimum Autoscale instances for App service plans, Application gateway based on current utilization
- Premium File shares underutilized provisioned capacity
-  Storage Account Access tier change for Blobs based on access pattern
- App service Plan rightsizing
- Azure Synapse Analytics rightsizing
- Azure SQL Managed instances rightsizing
- Azure Redis Cache rightsizing

5. Idle instances addition
- Idle VPN gateways 
- Unhealthy Backend pools for App gateway
- Idle VM for last 3months
- Storage not accessed for last 3months
- SQL Elastic Pools without DB's
- Disk Attached to stopped instances
- idle Load Balances with Low traffic

Please ignore if you see any ideas which are already implemented and duplicated.

## ℹ️ Additional context
TODO: Add any other context or screenshots about the change request, including alternative solutions you considered and why you ruled them out.

## 🙋‍♀️ Ask for the community

We could use your help:
1. Please vote this issue up (👍) to prioritize it.
8. Leave comments to help us solidify the vision.

**Comment by helderpinto** [comment]
Thanks for the suggestion, @rithinkumark1.

Regarding these:

> Idle VM for last 3months
> Storage not accessed for last 3months

As they require detailed historical usage data for a longer period, they are more in the scope of the [Azure Optimization Engine](https://aka.ms/AzureOptimizationEngine), which is going to be part of the FinOps Toolkit soon. Actually, AOE is able to flag idle VMs (for a configurable period, default is 30 days) with attached disks incurring costs.

**Comment by rithinkumark1** [comment]
Thanks @helderpinto We explored AOE before looks like some infra setup is required with additional cost involved hence, we did not move forward with the solution.

Any ETA for reviewing and implementing these suggestions if those are feasible?

**Comment by arthurclares** [comment]
Thank you for the suggestions Rithin.

Regarding this suggestion: 

> Flexibility to setup alerts in Optimization workbook. The goal is that we can setup email alert so that when a new orphaned resources/ recommendation is added to the workbook we can get an alert in our inbox. Right we need to manually check everyday if any new resources are added.
We are working to develop a logic app to alert for new idle resources.  @ro100e for awareness. 

Idle VPN GW is also implemented.

**Comment by rithinkumark1** [comment]
Thanks Arthur for the update. Is there any update on the implementation of the other suggestions?

**Comment by rithinkumark1** [comment]
Team, Addition to this request.
 Calculating the cost for idle backup seems to be nearly impossible as one can see only the RS vault specific charges. Is it possible to get the idle backup cost added in the workbook? If yes it would really provide a lot of value to a lot of customers.

**Comment by psilantropy** [comment]
> Cost breakout by orphaned resource category: It would be a nice feature to include the cost information for the orphaned resources. Right now we need to manually check the details in the cost management tool.

Hi @rithinkumark1 . Have you seen this orphan workbook before? 
https://github.com/dolevshor/azure-orphan-resources

It was updated a while ago to include $ markers when the resources cost money. It doesn't give you any costs and is just an icon indicator. But it definitely makes things stand out. Minor improvement :) 

There is also this with Azure Governance Visualizer which does what you ask, but it's a larger tool providing that small feature  - https://github.com/JulianHayward/Azure-MG-Sub-Governance-Reporting/releases/tag/v6_major_20220710_1

**Comment by rithinkumark1** [comment]
@psilantropy yes i have seen the oprhan workbook the $ addition looks good. I think this should be a great addition to the advisor optimization workbook as we use this as default for identifying orphaned resources.

The azure governance visualizer looks interesting i will have a look at this. Thanks

**Comment by ChiaChun0** [comment]
@flanakin and @arthurclares - I would like to contribute to this task for **Azure Databricks**, thank you.
